numba_rvsdg/core/datastructures/block_map.py

```diff
@@ -32,6 +33,29 @@ def __getitem__(self, index):
     def __contains__(self, index):
         return index in self.graph
 
+    def __iter__(self):
+        """Graph Iterator"""
+        # initialise housekeeping datastructures
+        to_visit, seen = [self.find_head()], []
```

Comment:
It would be more appropriate if we made `seen` a Set rather than a List, would allow faster checkins. 

`to_visit` could also be made into a set, however we'd lose the BFS property of this iteration, so it's a trade-off.